packages/zoe/src/types.js

```diff
@@ -0,0 +1,362 @@
+ * queries based on other information, we choose to omit it. For
+ * instance, `installationHandle` can be derived from
+ * `instanceHandle` and is omitted even though it is useful.
+ * @param {OfferHook<OC>} offerHook - a function that will be handed the
+ * offerHandle at the right time, and returns a contract-specific
+ * OfferOutcome which will be put in the OfferResultRecord.
+ * @param {string} inviteDesc
+ * @param {MakeInvitationOptions} [options]
+ * @returns {Promise<Invite<OC>>}
+ */
+
+/**
+ * @typedef MakeInvitationOptions
+ * @property {CustomProperties} [customProperties] - an object of
+ * information to include in the extent, as defined by the smart
+ * contract
+ */
+
+/**
+ * @template OC - the outcome type
+ * @callback OfferHook
+ * This function will be called with the OfferHandle when the offer
+ * is prepared. It should return a contract-specific "OfferOutcome"
+ * value that will be put in the OfferResultRecord.
+ * @param {OfferHandle} offerHandle
+ * @returns {OC}
+ */
+
+/**
+ * @callback AddNewIssuer
+ * Informs Zoe about an issuer and returns a promise for acknowledging
+ * when the issuer is added and ready.
+ * @param {Promise<Issuer>|Issuer} issuerP Promise for issuer
+ * @param {Keyword} keyword Keyword for added issuer
+ * @returns {Promise<IssuerRecord>} Issuer is added and ready
+ *
+ * @typedef {Record<string,function>} PublicAPI
+ *
+ * @callback InitPublicAPI
+ * Initialize the publicAPI for the contract instance, as stored by Zoe in
+ * the instanceRecord.
+ * @param {PublicAPI} publicAPI - an object whose methods are the API
+ * available to anyone who knows the instanceHandle
+ * @returns {void}
+ */
+
+/**
+ * @typedef {Object} VatAdmin
+ * A powerful object that can be used to terminate the vat in which a contract
+ * is running, to get statistics, or to be notified when it terminates. The
+ * VatAdmin object is only available to the contract from within the contract so
+ * that clients of the contract can tell (by getting the source code from Zoe
+ * using the instanceHandle) what use the contract makes of it. If they want to
+ * be assured of discretion, or want to know that the contract doesn't have the
+ * ability to call terminate(), Zoe makes this visible.
+ *
+ * @property {() => Object} done
+ * provides a promise that will be fullfilled when the contract is terminated.
+ * @property {() => undefined} terminate
+ * kills the vat in which the contract is running
+ * @property {() => Object} adminData
+ * provides some statistics about the vat in which the contract is running.
+ */
+
+/**
+ * @typedef {Object} CustomProperties
+ *
+ * @typedef {object} OfferRecord
+ * @property {OfferHandle} handle - opaque identifier, used as the table key
+ * @property {InstanceHandle} instanceHandle - opaque identifier for the instance
+ * @property {ProposalRecord} proposal - the offer proposal (including want, give, exit)
+ *
+ * @typedef {object} InstanceRecord
+ * @property {InstanceHandle} handle
```

Comment:
rather than using `omit` on the record, this should be changed to have the field name be instanceHandle.

Reply:
As above.